pkg/controller/keyvault/placeholder_pod.go

```diff
@@ -108,49 +110,92 @@ func (p *PlaceholderPodController) Reconcile(ctx context.Context, req ctrl.Reque
 	}
 	logger = logger.WithValues("deployment", dep.Name)
 
-	ing := &netv1.Ingress{}
-	ing.Name = util.FindOwnerKind(spc.OwnerReferences, "Ingress")
-	ing.Namespace = req.Namespace
-	logger = logger.WithValues("ingress", ing.Name)
-	if ing.Name != "" {
-		logger.Info("getting owner ingress")
-		if err = p.client.Get(ctx, client.ObjectKeyFromObject(ing), ing); err != nil {
-			return result, client.IgnoreNotFound(err)
+	if strings.HasPrefix(spc.Name, NginxNamePrefix) {
+		nic := &v1alpha1.NginxIngressController{}
+		nic.Name = util.FindOwnerKind(spc.OwnerReferences, "NginxIngressController")
+
+		if nic.Name != "" {
+			logger.Info("getting owner nginx ingress controller")
+			if err = p.client.Get(ctx, client.ObjectKeyFromObject(nic), nic); err != nil {
+				return result, client.IgnoreNotFound(err)
+			}
 		}
-	}
 
-	managed, err := p.ingressManager.IsManaging(ing)
-	if err != nil {
-		return result, fmt.Errorf("determining if ingress is managed: %w", err)
-	}
+		if nic.Name == "" || nic.Spec.IngressClassName == "" || nic.Spec.DefaultSSLCertificate.KeyVaultURI == "" {
```

Comment:
why nic.Spec.IngressClassName == ""?

Reply:
i don't think we need to check the nic.Spec.DefaultSSLCertificate.KeyVaultURI either